lib/ui/page/home/page/chat/gallery/controller.dart

```diff
@@ -0,0 +1,375 @@
+// Copyright © 2022-2025 IT ENGINEERING MANAGEMENT INC,
+//                       <https://github.com/team113>
+//
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the GNU Affero General Public License v3.0 as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License v3.0 for
+// more details.
+//
+// You should have received a copy of the GNU Affero General Public License v3.0
+// along with this program. If not, see
+// <https://www.gnu.org/licenses/agpl-3.0.html>.
+
+import 'dart:async';
+import 'dart:math';
+
+import 'package:dio/dio.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:get/get.dart';
+import 'package:flutter/widgets.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+import '/config.dart';
+import '/domain/model/attachment.dart';
+import '/domain/model/chat.dart';
+import '/domain/model/chat_item.dart';
+import '/domain/model/chat_item_quote.dart';
+import '/domain/repository/chat.dart';
+import '/domain/repository/paginated.dart';
+import '/domain/service/chat.dart';
+import '/l10n/l10n.dart';
+import '/routes.dart';
+import '/ui/page/home/page/chat/widget/chat_gallery.dart';
+import '/ui/page/player/controller.dart';
+import '/ui/page/player/view.dart';
+import '/ui/widget/text_field.dart';
+import '/util/message_popup.dart';
+
+export 'view.dart';
+
+/// Possible [Routes.chats][Routes.files] or [Routes.media] page screens.
+enum GalleryViewMode { files, media }
+
+/// Controller of the [Routes.chats] [Routes.files] or [Routes.media] page.
+class GalleryController extends GetxController {
+  GalleryController({
+    required this.chatService,
+    required this.chatId,
+    required this.viewMode,
+  });
+
+  /// ID of this [Chat].
+  final ChatId chatId;
+
+  /// Reactive state of the [Chat] this page is about.
+  final Rx<RxChat?> chat = Rx(null);
+
+  /// [ChatService] maintaining the [chat].
+  final ChatService chatService;
+
+  /// Current [GalleryView] view.
+  ///
+  /// May be:
+  /// - `viewMode.files`, meaning [GalleryView] must shown only files attachments.
+  /// - `viewMode.media`, meaning [GalleryView] must shown only media attachments.
+  final GalleryViewMode viewMode;
+
+  /// [GalleryView] [items] fetched from chat.
+  final RxList<Attachment> items = RxList([]);
+
+  /// Status of the [Chat] [GalleryView] fetching.
```

Comment:
What that status has to do with `GalleryView`. It's a status of the chat fetching, not a status of `GalleryView` chat fetching. `GalleryView` has nothing to do with chat being fetched.

And this question seems to araise for multiple fields in this file.